presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -52,6 +53,15 @@ public RuleAssert assertThat(Rule rule)
         return new RuleAssert(metadata, session, rule);
     }
 
+    public void assertInTransaction(Rule rule, Consumer<RuleAssert> ruleAssertion)
```

Comment:
I am sorry I wasn't giving detailed information. PatternMatcher checks the metadata. Without transaction,  tableScan node will throw when matching with a pattern:

```
java.lang.IllegalStateException: Not in a transaction

at com.google.common.base.Preconditions.checkState(Preconditions.java:444)
at com.facebook.presto.Session.getRequiredTransactionId(Session.java:187)
at com.facebook.presto.metadata.MetadataManager.getCatalogMetadata(MetadataManager.java:895)
at com.facebook.presto.metadata.MetadataManager.getMetadata(MetadataManager.java:910)
at com.facebook.presto.metadata.MetadataManager.getTableMetadata(MetadataManager.java:336)
at com.facebook.presto.sql.planner.assertions.TableScanMatcher.detailMatches(TableScanMatcher.java:62)
at com.facebook.presto.sql.planner.assertions.PlanMatchPattern.detailMatches(PlanMatchPattern.java:339)
at com.facebook.presto.sql.planner.assertions.PlanMatchingVisitor.matchLeaf(PlanMatchingVisitor.java:132)
at com.facebook.presto.sql.planner.assertions.PlanMatchingVisitor.visitPlan(PlanMatchingVisitor.java:93)
at com.facebook.presto.sql.planner.assertions.PlanMatchingVisitor.visitPlan(PlanMatchingVisitor.java:33)
at com.facebook.presto.sql.planner.plan.PlanVisitor.visitTableScan(PlanVisitor.java:70)
at com.facebook.presto.sql.planner.plan.TableScanNode.accept(TableScanNode.java:135)
```